### ISSUE TITLE
Cannot clear firmware passcode using MDM Command

### ISSUE THREAD

**Issue by tperfitt** [micromdm/micromdm#703]
# What version of `micromdm` are you using?
1.7

When sending the MDM command to clean the firmware passcode, the spec says:

NewPasscode:  Pass an empty string to clear the firmware password

However, MicroMDM doesn't send a empty string for NewPassword and the mdm client gives back this error :

			<key>ErrorCode</key>
			<integer>-104</integer>
			<key>ErrorDomain</key>
			<string>FirmwarePasswordService</string>
			<key>LocalizedDescription</key>
			<string>'NewPassword' not provided</string>

I believe the issue is that the NewPasscode is set to omitempty:

	NewPassword     string `plist:",omitempty" json:"new_password,omitempty"`

Please update so we can send an empty string to remove the firmware passcode.

**Comment by jessepeterson** [comment]
If you want to test the change (and maybe PR it) that'd be helpful. If you were up for it even a unit test for this would be helpful, too. :) I think the change would look something like this:

```diff
diff --git a/mdm/mdm/command.go b/mdm/mdm/command.go
index ab3d6a4..747ca54 100644
--- a/mdm/mdm/command.go
+++ b/mdm/mdm/command.go
@@ -242,7 +242,7 @@ type ManagedApplicationFeedback struct {
 
 type SetFirmwarePassword struct {
        CurrentPassword string `plist:",omitempty" json:"current_password,omitempty"`
-       NewPassword     string `plist:",omitempty" json:"new_password,omitempty"`
+       NewPassword     string `json:"new_password"`
        AllowOroms      bool   `plist:",omitempty" json:"allow_oroms,omitempty"`
 }
```

**Comment by tomaswallentinus** [comment]
I tried the change but the response from Apple is:
`<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CommandUUID</key>
	<string>39bbe1c3-3925-46f9-a38b-e149b89c31a2</string>
	<key>ErrorChain</key>
	<array>
		<dict>
			<key>ErrorCode</key>
			<integer>-106</integer>
			<key>ErrorDomain</key>
			<string>FirmwarePasswordService</string>
			<key>LocalizedDescription</key>
			<string>GetFirmwarePasswordMode returned: −4</string>
		</dict>
	</array>
	<key>Status</key>
	<string>Error</string>
	<key>UDID</key>
	<string>EC30E9CF-5BC0-5A65-BB33-F637F6921BA7</string>
</dict>
</plist>
`

**Comment by jessepeterson** [comment]
Addressed in #743.
